Hermes/appscale/hermes/stats/producers/rabbitmq_stats.py

```diff
@@ -75,3 +93,44 @@ def get_current():
     logging.info('Prepared RabbitMQ node stats in '
                  '{elapsed:.1f}s.'.format(elapsed=time.time()-start))
     return snapshot
+
+
+class PushQueueStatsSource(object):
+  """ Fetches push queue stats. """
+
+  first_run = True
+
+  @staticmethod
+  def get_current():
+    """ Retrieves push queue stats.
+
+    Returns:
+      An instance of PushQueueStatsSnapshot.
+    """
+    start = time.time()
+
+    url = 'http://localhost:{}{}'.format(API_PORT, QUEUES_API)
+    creds = base64.b64encode(':'.join([USER, PASS]))
+    headers = {'Authorization': 'Basic {}'.format(creds)}
+    client = HTTPClient()
+    try:
+      response = client.fetch(url, headers=headers)
+    except Exception as error:
+      raise APICallFailed('Call to {} failed: {}'.format(url, error))
+
+    try:
+      queues_info = json.loads(response.body)
+    except ValueError:
+      raise APICallFailed('Invalid response from '
+                          '{}: {}'.format(url, response.body))
+
+    queue_stats = [
+      PushQueueStats(name=queue['name'], messages=queue['messages'])
```

Comment:
Is "messages" number of messages?

Reply:
Yes. It's the number of messages waiting to be consumed.